pkg/auto/notificationsemail/config/root.go

```diff
@@ -0,0 +1,217 @@
+		err = fmt.Errorf("destination.recipients is empty")
+		return
+	}
+	// defaults
+	if cfg.Destination.SendTime == nil {
+		cfg.Destination.SendTime = DefaultSendTime
+	}
+
+	// validate email addresses
+	parsed, err := cfg.Destination.Parse()
+	if err != nil {
+		return
+	}
+	cfg.Destination.Parsed = parsed
+	for i, recipient := range cfg.Destination.To {
+		_, err = mail.ParseAddress(recipient)
+		if err != nil {
+			err = fmt.Errorf("destination.recipients[%d] is invalid: %w", i, err)
+			return
+		}
+	}
+	return
+}
+
+func (d Destination) Parse() (*ParsedDestination, error) {
+	p := &ParsedDestination{}
+	var err error
+	p.Addr = d.Addr()
+	p.Username = d.Username
+	p.Password, err = d.Password.Read()
+	if err != nil {
+		return nil, fmt.Errorf("destination.password: %w", err)
+	}
+
+	p.From, err = mail.ParseAddress(d.From)
+	if err != nil {
+		return nil, fmt.Errorf("destination.from: %w", err)
+	}
+	if p.Username == "" {
+		p.Username = p.From.Address
+	}
+
+	for i, to := range d.To {
+		a, err := mail.ParseAddress(to)
+		if err != nil {
+			return nil, fmt.Errorf("destination.to[%d]: %w", i, err)
+		}
+		p.To = append(p.To, a)
+	}
+	sort.Slice(p.To, func(i, j int) bool {
+		return p.To[i].Address < p.To[j].Address
+	})
+
+	p.SubjectTemplate, err = d.ReadSubjectTemplate()
+	if err != nil {
+		return nil, fmt.Errorf("destination.subjectTemplate: %w", err)
+	}
+	p.BodyTemplate, err = d.ReadBodyTemplate()
+	if err != nil {
+		return nil, fmt.Errorf("destination.bodyTemplate: %w", err)
+	}
+	return p, nil
+}
+
+// Addr returns the combination of Host and Port, taking defaults into account.
+// Suitable for smtp.Dial.
+func (d Destination) Addr() string {
+	p := d.Port
+	if p == 0 {
+		p = 587
+	}
+	return net.JoinHostPort(d.Host, strconv.Itoa(p))
+}
+
+func (d Destination) ReadSubjectTemplate() (*template.Template, error) {
+	s, err := d.SubjectTemplate.Read()
+	if err != nil {
+		return nil, err
+	}
+	if s == "" {
+		s = DefaultEmailSubject
+	}
+	return template.New("subject").Parse(s)
+}
+
+func (d Destination) ReadBodyTemplate() (*template.Template, error) {
+	s, err := d.BodyTemplate.Read()
+	if err != nil {
+		return nil, err
+	}
+	if s == "" {
+		s = DefaultEmailBody
+	}
+	return template.New("body").
+		Funcs(template.FuncMap{
+			"printTime": func(t any) string {
+				format := time.Stamp
+				switch t := t.(type) {
+				case time.Time:
+					return t.Format(format)
+				case *time.Time:
+					return t.Format(format)
+				case *timestamppb.Timestamp:
+					return t.AsTime().Format(format)
+				case nil:
+					return ""
+				default:
+					return fmt.Sprintf("%v", t)
+				}
+			},
+		}).
+		Parse(s)
+}
+
+func (d *AttachmentCfg) AttachFile(name string, b []byte) {
+	d.AttachmentName = name
+	d.Attachment = b
+}
+
+const DefaultEmailSubject = `Smart Core Notifications Monthly Report {{.Now.Format "Jan 02, 2006"}}`
+const DefaultEmailBody = `<html lang="en">
+<head>
+  <title>One Centenary Way - Smart Core Notifications</title>
```

Comment:
Not great having a client name in sc-bos automation. Is there a way to have this templated and pulled in via config?